src/coreclr/vm/interoplibinterface_comwrappers.cpp

```diff
@@ -672,1402 +48,682 @@ namespace
-            flags |= CreateObjectFlags::CreateObjectFlags_UniqueInstance;
-
-        // Passing NULL as the ComWrappers implementation indicates using the globally registered instance
-        return TryGetOrCreateObjectForComInstanceInternal(
-            NULL /*comWrappersImpl*/,
-            g_marshallingGlobalInstanceId,
-            identity,
-            NULL,
-            (CreateObjectFlags)flags,
-            ComWrappersScenario::MarshallingGlobalInstance,
-            NULL /*wrapperMaybe*/,
-            objRef);
-    }
-}
+        if (handle == nullptr)
+            return false;
 
-extern "C" void QCALLTYPE ComWrappers_SetGlobalInstanceRegisteredForTrackerSupport(INT64 id)
-{
-    QCALL_CONTRACT_NO_GC_TRANSITION;
+        ::OBJECTHANDLE objectHandle = static_cast<::OBJECTHANDLE>(handle);
 
-    _ASSERTE(g_trackerSupportGlobalInstanceId == ComWrappersNative::InvalidWrapperId && id != ComWrappersNative::InvalidWrapperId);
-    g_trackerSupportGlobalInstanceId = id;
-}
+        OBJECTREF obj = ObjectFromHandle(objectHandle);
 
-bool GlobalComWrappersForTrackerSupport::IsRegisteredInstance(INT64 id)
-{
-    return g_trackerSupportGlobalInstanceId != ComWrappersNative::InvalidWrapperId
-        && g_trackerSupportGlobalInstanceId == id;
-}
+        if (obj == nullptr)
+            return false;
 
-bool GlobalComWrappersForTrackerSupport::TryGetOrCreateComInterfaceForObject(
-    _In_ OBJECTREF instance,
-    _Outptr_ void** wrapperRaw)
-{
-    CONTRACTL
-    {
-        THROWS;
-        MODE_COOPERATIVE;
+        return GCHeapUtilities::GetGCHeap()->IsPromoted(OBJECTREFToObject(obj));
     }
-    CONTRACTL_END;
 
-    if (g_trackerSupportGlobalInstanceId == ComWrappersNative::InvalidWrapperId)
-        return false;
+    void SetGlobalPeggingState(_In_ bool state) noexcept
+    {
+        CONTRACTL
+        {
+            NOTHROW;
+            GC_NOTRIGGER;
+            MODE_ANY;
+        }
+        CONTRACTL_END;
 
-    // Passing NULL as the ComWrappers implementation indicates using the globally registered instance
-    return TryGetOrCreateComInterfaceForObjectInternal(
-        NULL,
-        g_trackerSupportGlobalInstanceId,
-        instance,
-        CreateComInterfaceFlags::CreateComInterfaceFlags_TrackerSupport,
-        ComWrappersScenario::TrackerSupportGlobalInstance,
-        wrapperRaw);
-}
+        BOOL newState = state ? TRUE : FALSE;
+        VolatileStore(&g_isGlobalPeggingOn, newState);
+    }
 
-bool GlobalComWrappersForTrackerSupport::TryGetOrCreateObjectForComInstance(
-    _In_ IUnknown* externalComObject,
-    _Out_ OBJECTREF* objRef)
-{
-    CONTRACTL
+    TryInvokeICustomQueryInterfaceResult TryInvokeICustomQueryInterface(
+        _In_ InteropLib::OBJECTHANDLE handle,
+        _In_ REFGUID iid,
+        _Outptr_result_maybenull_ void** obj) noexcept
     {
-        THROWS;
-        MODE_COOPERATIVE;
-    }
-    CONTRACTL_END;
+        CONTRACTL
+        {
+            NOTHROW;
+            MODE_ANY;
```

Comment:
Why any?

Reply:
This is `ANY` in main today, I didn't change it.